### DISCUSSION TITLE
cluster issue

### DISCUSSION THREAD

**Discussion by lucky-zhao** [minio/minio#13844] [Q&A] [answered]
I deployed a cluster with etcd
IP:205 and 207
```
export MINIO_ETCD_ENDPOINTS="http://10.5.15.205:2379,http://10.5.15.207:2379,http://10.5.15.208:2379"

export MINIO_PUBLIC_IPS=10.5.15.205,10.5.15.207
export MINIO_DOMAIN=domain.com
export MINIO_ROOT_USER=minioadmin
export MINIO_ROOT_PASSWORD=minioadmin
export MINIO_VOLUMS="/tmp/data"
./minio server --console-address ":9001" http://10.5.15.205/tmp/minio_data/data{1...4}  http://10.5.15.207/tmp/minio_data/data{1...4}
```

Then I expand the cluster

```

export MINIO_ETCD_ENDPOINTS="http://10.5.15.205:2379,http://10.5.15.207:2379,http://10.5.15.208:2379"

export MINIO_PUBLIC_IPS=10.5.15.201,10.5.15.208
export MINIO_DOMAIN=domain.com
export MINIO_ROOT_USER=minioadmin
export MINIO_ROOT_PASSWORD=minioadmin
./minio server --console-address ":9001" http://10.5.15.201/tmp/minio_data/data{1...4}  http://10.5.15.208/tmp/minio_data/data{1...4}
```
it is ok,However, when I visit the page with IP address 201, I can see the bucket created by the cluster with IP address 205. When I click in, an error exception will be reported:The specified bucket does not exist。
 Different clusters can't see each other's buckets. What's wrong with me?

![image](https://user-images.githubusercontent.com/45654758/144846658-4604d717-6f3e-48b4-b28c-340b783330f8.png)

**Comment by lucky-zhao** [comment]
version: minio/minio:latest

**Comment by vadmeste** [comment]
@lucky-zhao can you try `mc ls <alias>` here ?

**Comment by lucky-zhao** [comment]
I used `mc ls myminio` in the first cluster, but I could see the buckets of another cluster，I don't think it should be，But I don't know what's wrong

**Comment by lucky-zhao** [comment]
![image](https://user-images.githubusercontent.com/45654758/144851606-81ff4228-1b62-464b-91b7-55ceb712f9c3.png)

The two barrels marked by the picture should not appear

**Comment by klauspost** [comment]
@lucky-zhao You are creating separate clusters, you are not expanding.

**Comment by vadmeste** [comment]
@lucky-zhao you are actually activating federation feature, which means you are linking the two clusters into one. That's why you are seeing buckets from both clusters in one. 

What you are trying to achieve and why you are setting this environment variable, MINIO_PUBLIC_IPS ?

**Comment by lucky-zhao** [comment]
> Contributor



> @lucky-zhao you are actually activating federation feature, which means you are linking the two clusters into one. That's why you are seeing buckets from both clusters in one.
> 
> What you are trying to achieve and why you are setting this environment variable, MINIO_PUBLIC_IPS ?

I created a cluster with IP: 205 and IP: 207 according to the tutorial and recorded the information to etcd. Then I expanded a cluster with IP addresses 201 and 208. What I want to achieve is that 205 and 207 are one cluster and 201 and 208 are another cluster. They are invisible to each other. I want to try Minio capacity expansion

**Comment by klauspost** [comment, marked as answer]
Your setup is wrong. You have created 2 server pools for your initial cluster. This is a bad setup, but it will work.

Since you don't have consecutive IPs, you should use DNS-based setup. The existing setup is bad since each server is a server pool. You can add extend it by adding the two other servers as 2 more server pools. However, it will be bad since you don't get any safety if any sever dies.

I propose you re-read the docs and set up your cluster again: https://docs.min.io/minio/baremetal/installation/deploy-minio-distributed.html#deploy-minio-distributed and https://docs.min.io/minio/baremetal/installation/expand-minio-distributed.html

As your servers are set up they are behaving as expected.
